test/gtest/aca_test_dhcp.cpp

```diff
@@ -180,3 +224,341 @@ TEST(dhcp_message_test_cases, get_options_valid)
+  subnetConfig_GatewayBuilder->set_mac_address(subnet1_gw_mac);
+  SubnetConiguration_builder->set_allocated_gateway(subnetConfig_GatewayBuilder);
+
+  // create a new port 1 in demo mode
+  GoalStateOperationReply gsOperationalReply;
+
+  overall_rc = Aca_Comm_Manager::get_instance().update_goal_state(
+          GoalState_builder, gsOperationalReply);
+  ASSERT_EQ(overall_rc, EXIT_SUCCESS);
+
+  // create a new port 2 in demo mode
+  DHCPConfiguration_builder->set_mac_address(vmac_address_3);
+  DHCPConfiguration_builder->set_ipv4_address(vip_address_3);
+  DHCPConfiguration_builder->set_port_host_name(port_name_3);
+
+  SubnetConiguration_builder->set_id(subnet_id_2);
+  SubnetConiguration_builder->set_cidr(subnet2_cidr);
+  SubnetConiguration_builder->set_tunnel_id(124);
+
+  subnetConfig_GatewayBuilder->set_ip_address(subnet2_gw_ip);
+  subnetConfig_GatewayBuilder->set_mac_address(subnet2_gw_mac);
+  SubnetConiguration_builder->set_allocated_gateway(subnetConfig_GatewayBuilder);
+
+  overall_rc = Aca_Comm_Manager::get_instance().update_goal_state(
+          GoalState_builder, gsOperationalReply);
+  ASSERT_EQ(overall_rc, EXIT_SUCCESS);
+
+  // send goal state ok, now create a docker instance test
+  // con1
+  overall_rc = Aca_Net_Config::get_instance().execute_system_command(
+          "docker run -itd --cap-add=NET_ADMIN --name con1 --net=none alpine sh");
+  EXPECT_EQ(overall_rc, EXIT_SUCCESS);
+
+  cmd_string = "ovs-docker add-port br-int eth1 con1 --macaddress=" + vmac_address_1;
+  overall_rc = Aca_Net_Config::get_instance().execute_system_command(cmd_string);
+  EXPECT_EQ(overall_rc, EXIT_SUCCESS);
+  overall_rc = EXIT_SUCCESS;
+
+  cmd_string = "ovs-docker set-vlan br-int eth1 con1 1";
+  overall_rc = Aca_Net_Config::get_instance().execute_system_command(cmd_string);
+  EXPECT_EQ(overall_rc, EXIT_SUCCESS);
+  overall_rc = EXIT_SUCCESS;
+
+  cmd_string = "docker exec con1 udhcpc -i eth1";
+  Aca_Net_Config::get_instance().execute_system_command(cmd_string);
+  EXPECT_EQ(overall_rc, EXIT_SUCCESS);
+  overall_rc = EXIT_SUCCESS;
+
+  // con2
+  overall_rc = Aca_Net_Config::get_instance().execute_system_command(
+          "docker run -itd --cap-add=NET_ADMIN --name con2 --net=none alpine sh");
+  EXPECT_EQ(overall_rc, EXIT_SUCCESS);
+
+  cmd_string = "ovs-docker add-port br-int eth1 con2 --macaddress=" + vmac_address_3;
+  overall_rc = Aca_Net_Config::get_instance().execute_system_command(cmd_string);
+  EXPECT_EQ(overall_rc, EXIT_SUCCESS);
+
+  cmd_string = "ovs-docker set-vlan br-int eth1 con2 2";
+  overall_rc = Aca_Net_Config::get_instance().execute_system_command(cmd_string);
+  EXPECT_EQ(overall_rc, EXIT_SUCCESS);
+
+  cmd_string = "docker exec con2 udhcpc -i eth1";
+  Aca_Net_Config::get_instance().execute_system_command(cmd_string);
+  EXPECT_EQ(overall_rc, EXIT_SUCCESS);
+
+  // enable ip netns for router
+  // add dhcp router namespace
+  cmd_string = "ip netns add " + dhcp_test_router_namespace;
```

Comment:
I am curious, how does this dhcp_test_router_namespace routes packets from one subnet to another subnet? Does it happen automatically?

Reply:
Yes, It seems like the neutron router.